code/modules/mob/living/carbon/human/diona_gestalt.dm

```diff
@@ -359,6 +368,69 @@
 	visible_message(SPAN_WARNING("\The [src] quivers slightly, then splits apart with a wet slithering noise."))
 	qdel(src)
 
+// Miscellaneous abilities
+//================================
+
+/**
+This verb lets the Diona anchor themselves to the ground, giving them magboot properties.
+Slows you down while being used, and cannot be used below 50 nutrition and 20 energy.
+*/
+/mob/living/carbon/human/proc/root_to_ground()
+	set name = "Extend Roots"
+	set desc = "Extend your roots to keep yourself upright in case of pressure changes."
+	set category = "Abilities"
+
+	if(use_check_and_message(src))
+		return
+	if(nutrition <= 50)
+		to_chat(src, SPAN_WARNING("You lack nutrition to perform this action!"))
+		return
+	if(DS.stored_energy <= 20)
+		to_chat(src, SPAN_WARNING("You lack energy to perform this action!"))
+		return
+	if(last_special > world.time)
+		to_chat(src, SPAN_WARNING("You've used an ability too recently! Wait a bit."))
+		return
+	if(!has_organ(BP_R_FOOT) && !has_organ(BP_L_FOOT))
+		to_chat(src, SPAN_WARNING("You have no supporting limbs to extend your roots from!"))
+		return
+
+	last_special = world.time + 2 SECONDS
+
+	// Checks whether we want to disable or enable the ability with this call.
+	if(HAS_TRAIT(src, TRAIT_SHOE_GRIP))
```

Comment:
use HAS_TRAIT_FROM and check for TRAIT_SOURCE_SPECIESVERB or whatever the tag i added was